app/models/ems_event.rb

```diff
@@ -146,6 +146,15 @@ def self.process_vm_in_event!(event, options = {})
         event[:vm_name] ||= vm.name
       end
     end
+
+    event_type = event[:event_type]
+    # both event types are Rhev specific
+    if (event_type == 'USER_ADD_VM' || event_type == 'USER_ADD_VM_FINISHED_SUCCESS') && event[:vm_or_template_id].nil?
+      new_vm = ManageIQ::Providers::Redhat::InfraManager::Vm.create_from_event(event)
```

Comment:
I apologise for barging in, but I am just curious whether there is no other way to make this provider specific code separated from this model. From what I understand the two events are (currently) used only by RedHat provider. However, the code that was added here actually overrides whatever previous lines put in the event object as it basically sets the id and name again.

So I would either change this code to check for the specific event type earlier in this method or, even better, move Rhev specific code to somewhere else, if possible.